pkg/validations/input.go

```
@@ -40,3 +43,25 @@ func KubeConfigExists(dir, clusterName string, kubeConfigFileOverride string, ku
 	}
 	return false
 }
+
+func ValidateTaintsSupport(ctx context.Context, clusterSpec *cluster.Spec) error {
+	if !features.IsActive(features.TaintsSupport()) {
+		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 ||
+			len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].Taints) > 0 {
+			return fmt.Errorf("Taints feature is not enabled. Environment variable TAINTS_SUPPORT needs to be set to true.")
+		}
+	} else if len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].Taints) > 0 {
+		invalidWorkerNodeGroupTaints := false
+		for _, slice := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].Taints {
+			if slice.Effect == "NoExecute" || slice.Effect == "NoSchedule" {
+				invalidWorkerNodeGroupTaints = true
```

Why not just return the error from here?